### DISCUSSION TITLE
set active className on Link in _document

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [vercel/next.js#15872] [Help]
Hello,
I'm looking for a way to set active tab style on Link elements defined in the custom _document. What I found online is ways that use the router to get the pathname and set the active class based on the href if they're equal.

So far the router always returned null, and I read in the docs that _document only runs on the server.
Do you think there might be workarounds for this??

**Comment by akellbl4** [comment]
Why do you want to do it in `_document.js`? I think`_app.js` could be a better place for it. `_app.js` has access to the router.

**Reply by omar-dulaimi** [reply]
You are right, I ended up doing dom manipulation there and now everything works except for the pathname of homepage, which has the value of  "/". It seems like it doesn't go through _app.js  at all. Any thoughts about that?

**Reply by akellbl4** [reply]
Does the homepage have a diffrent path from "/"? Usually, it has "/" :)

**Reply by omar-dulaimi** [reply]
No, it's the same path. And I think that I've just found out the problem, it didn't get called in _app.js because it's an AMP page.
